R/caretStack.R

```diff
@@ -165,16 +266,24 @@ predict.caretStack <- function(
     excluded_class_id <- 0L
   }
 
+  # Make sure original features saved in the training data are present in the newdata
+  if (!is.null(newdata) && !is.null(object$original_features_included)) {
+    check_newdata_features(newdata, object$original_features_included)
+  }
+
   # Get predictions from the submodels on the new data
   # We need theres if there's newdata, for passing the base model predictions to the stack model
   # We also need these if we're calculting standard errors for the predictions
-  sub_model_preds <- if (!is.null(newdata) || se) {
-    stats::predict(
+  if (!is.null(newdata) || se) {
+    sub_model_preds <- stats::predict(
       object$models,
       newdata = newdata,
       verbose = verbose,
       excluded_class_id = object[["excluded_class_id"]]
     )
+    if (!is.null(object$original_features_included)) {
+      sub_model_preds <- cbind(sub_model_preds, newdata[, object$original_features_included])
```

Comment:
this looks good.  If we change the parameter in caretStack to be a single character vector, the logic in caretStack can be as simple as this too.